.github/workflows/backup-artifact.yml

```diff
@@ -1,17 +1,31 @@
 name: Build:Main
 on:
   push:
-    branches: [ master, test-ready ]
+    branches: [ master, test-ready, mac_build ]
 jobs:
   build-libraries:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
     env:
       CLIENTREPO: ${GITHUB_WORKSPACE}/client
     steps:
     - name: install AWS requirements
       id: install-aws-requirements
       run: |
-        sudo apt install -y libcurl4-openssl-dev libssl-dev uuid-dev zlib1g-dev libpulse-dev
+        if [ "$RUNNER_OS" == "Linux" ]; then
+          sudo apt install -y libcurl4-openssl-dev libssl-dev uuid-dev zlib1g-dev libpulse-dev
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+          brew update
+          brew install openssl@1.1 readline
+          brew install cmake
+        else
+          echo "$RUNNER_OS not supported"
+          exit 1
+        fi
```

Comment:
The aws cache below does not look correct.  If you changed the `ref:` to the aws SDK wouldn't it still have a cache hit?  The example [here](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows) hashes the files to make sure they are indeed the same.